packages/flutter_tools/lib/src/commands/build_ios.dart

```diff
@@ -51,20 +51,14 @@ class BuildIOSCommand extends FlutterCommand {
 
     printStatus('Building the application for $logTarget.');
 
-    Directory buildDir = new Directory(path.join('build', 'ios_$logTarget'));
```

Comment:
Yeah. The snapshotter was expecting the generated dylib containing precompiled code to be in the within the .generated folder. I am going to fix that and revert this removal then.